kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -124,7 +126,9 @@ private[kyuubi] class EngineRef private(conf: KyuubiConf, user: String, sessionI
   }
 
   private def get(zkClient: CuratorFramework): Option[(String, Int)] = {
-    getServerHost(zkClient, engineSpace)
+    getServiceNodesInfo(zkClient, engineSpace, silent = true)
```

Comment:
https://github.com/apache/incubator-kyuubi/pull/935/files#diff-6c518a53d95d41547d6677ef6f3b7557f1e52a93a8c81ab547d95ca19d601f53R135-R137

we shouldn't filter by id here

Reply:
Two method calls to ServiceDiscovery
```
(No filter )    KyuubiSessionImpl -> getOrCreate -> getServerHost
(Filter by id)  EngineRef -> create -> getEngineBySessionId -> getServiceNodesInfo
```

Reply:
`EngineRef -> create -> (blocked by lock) -> (obtain lock) -> (try to get engine by another success) -> get OR actually create`

Reply:
right, got your point, we need filter id here, and already rename `get` to `getEngineBySessionId`.